pkg/crc/machine/bundle/copier.go

```diff
@@ -0,0 +1,166 @@
+	copier.copiedBundle.cachedPath = bundlePath
+
+	return &copier, nil
+}
+
+func (copier *Copier) Cleanup() error {
+	return os.RemoveAll(copier.copiedBundle.cachedPath)
+}
+
+func (copier *Copier) resolvePath(filename string) string {
+	return copier.copiedBundle.resolvePath(filename)
+}
+
+func (copier *Copier) CopyPrivateSSHKey(srcPath string) error {
+	sshKeyFileName := filepath.Base(copier.srcBundle.GetSSHKeyPath())
+	destPath := copier.resolvePath(sshKeyFileName)
+	if err := crcos.CopyFileContents(srcPath, destPath, 0400); err != nil {
+		return err
+	}
+	// copier.srcBundle.ClusterInfo.SSHPrivateKeyFile = sshKeyFileName
+	return nil
+}
+
+func (copier *Copier) SetKubeAdminPassword(kubePassword string) error {
+	kubeAdminPasswordFileName := copier.srcBundle.ClusterInfo.KubeadminPasswordFile
+	path := copier.resolvePath(kubeAdminPasswordFileName)
+	if err := ioutil.WriteFile(path, []byte(kubePassword), 0600); err != nil {
+		return err
+	}
+	// copier.copiedBundle.ClusterInfo.KubeadminPasswordFile = kubeAdminPasswordFileName
+	return nil
+}
+
+func (copier *Copier) CopyKubeConfig() error {
+	kubeConfigFileName := filepath.Base(copier.srcBundle.GetKubeConfigPath())
+	srcPath := copier.srcBundle.GetKubeConfigPath()
+	destPath := copier.resolvePath(kubeConfigFileName)
+	if err := crcos.CopyFileContents(srcPath, destPath, 0640); err != nil {
+		return err
+	}
+	// copier.copiedBundle.ClusterInfo.KubeConfig = kubeConfigFileName
+	return nil
+}
+
+func (copier *Copier) CopyOcExecutable() error {
+	srcPath := copier.srcBundle.GetOcPath()
+	destPath := copier.resolvePath(constants.OcExecutableName)
+	if err := crcos.CopyFileContents(srcPath, destPath, 0640); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (copier *Copier) SetDiskImage(path string) error {
+	fileInfo, err := getFileInfo(path)
+	if err != nil {
+		return err
+	}
+	diskImage := DiskImage{
+		File:   *fileInfo,
+		Format: copier.srcBundle.GetDiskImageFormat(),
+	}
+
+	copier.copiedBundle.Storage.DiskImages = []DiskImage{diskImage}
+
+	return nil
+}
+
+func (copier *Copier) GenerateBundle(bundleName string) error {
+	if err := copier.copiedBundle.verify(); err != nil {
+		return err
+	}
+
+	// update bundle info
+	copier.copiedBundle.BuildInfo.BuildTime = time.Now().String()
+
+	// Create the metadata json for custom bundle
+	bundleContent, err := json.MarshalIndent(copier.copiedBundle, "", " ")
+	if err != nil {
+		return err
+	}
+	err = ioutil.WriteFile(copier.resolvePath("crc-bundle-info.json"), bundleContent, 0600)
+	if err != nil {
+		return fmt.Errorf("error copying bundle metadata  %w", err)
+	}
+
+	logging.Infof("Compressing %s...", copier.copiedBundle.cachedPath)
+	if err := compress.Compress(copier.copiedBundle.cachedPath, fmt.Sprintf("%s%s", bundleName, constants.DefaultBundleExtension)); err != nil {
+		return err
+	}
+	return nil
+}
+
+func sha256sum(path string) (string, error) {
+	logging.Infof("Generating sha256sum for %s...", filepath.Base(path))
```

Comment:
Maybe just `debug`?

Reply:
@cfergeau it takes fair amount of time so `Info` is preferable than `Debug`.